From bfde6d171ec90b1d797228e517e21dd4db317097 Mon Sep 17 00:00:00 2001
From: Richard Liu <r66033@freescale.com>
Date: Thu, 19 Dec 2013 15:54:04 +0800
Subject: [PATCH 23/23] ENGR00292763 fix virtual display output format issue

on KitKat4.4.2 virtual display output format will be set to
HAL_PIXEL_FORMAT_IMPLEMENTATION_DEFINED if the memory usage
not set SW_READ or SW_WRITE flag, and normally this format
will be redefine to YUV format according to the platfrom, but
mx6x hwcomposer don't suppport YUV format output, we remove
such logic code and just use sinkFormat.

Signed-off-by: Richard Liu <r66033@freescale.com>
---
 .../DisplayHardware/VirtualDisplaySurface.cpp        | 20 ++++++--------------
 1 file changed, 6 insertions(+), 14 deletions(-)

diff --git a/services/surfaceflinger/DisplayHardware/VirtualDisplaySurface.cpp b/services/surfaceflinger/DisplayHardware/VirtualDisplaySurface.cpp
index be5cf4a..f3170a9 100644
--- a/services/surfaceflinger/DisplayHardware/VirtualDisplaySurface.cpp
+++ b/services/surfaceflinger/DisplayHardware/VirtualDisplaySurface.cpp
@@ -67,20 +67,12 @@ VirtualDisplaySurface::VirtualDisplaySurface(HWComposer& hwc, int32_t dispId,
     sink->query(NATIVE_WINDOW_WIDTH, &sinkWidth);
     sink->query(NATIVE_WINDOW_HEIGHT, &sinkHeight);
 
-    // Pick the buffer format to request from the sink when not rendering to it
-    // with GLES. If the consumer needs CPU access, use the default format
-    // set by the consumer. Otherwise allow gralloc to decide the format based
-    // on usage bits.
-    int sinkUsage;
-    sink->query(NATIVE_WINDOW_CONSUMER_USAGE_BITS, &sinkUsage);
-    if (sinkUsage & (GRALLOC_USAGE_SW_READ_MASK | GRALLOC_USAGE_SW_WRITE_MASK)) {
-        int sinkFormat;
-        sink->query(NATIVE_WINDOW_FORMAT, &sinkFormat);
-        mDefaultOutputFormat = sinkFormat;
-    } else {
-        mDefaultOutputFormat = HAL_PIXEL_FORMAT_IMPLEMENTATION_DEFINED;
-    }
-    mOutputFormat = mDefaultOutputFormat;
+    // Normally HAL_PIXEL_FORMAT_IMPLEMENTATION_DEFINED will redefine to some
+    // YUV format according to the platfrom, but mx6x hwcomposer don't suppport
+    // YUV format output so here set the output format to sinkFormat. 
+    int sinkFormat;
+    sink->query(NATIVE_WINDOW_FORMAT, &sinkFormat);
+    mOutputFormat = mDefaultOutputFormat = sinkFormat;
 
     ConsumerBase::mName = String8::format("VDS: %s", mDisplayName.string());
     mConsumer->setConsumerName(ConsumerBase::mName);
-- 
1.8.0

